include/boost/url/rfc/detail/impl/hier_part_rule.ipp

```diff
@@ -65,7 +65,13 @@ parse(
         t.authority = *rv;
         t.has_authority = true;
     }
-    if(it == end)
+    // the authority requires an absolute path
+    // or an empty path
+    if(it == end ||
+       (t.has_authority &&
+         (*it != '/' &&
+          *it != '?' &&
+          *it != '#')))
```

Comment:
needs a test or two obviously

Reply:
The spacing is weird, I would have written it as
```
    if(it == end || (
        t.has_authority && (
            *it != '/' &&
            *it != '?' &&
            *it != '#')))
```

This way everything is aligned on tab stops without added spaces

Reply:
Oh! :open_mouth: Done.

I forgot to bring any tests from the `format` branch. I came up with a simpler test here.

Reply:
I like simpler tests